### ISSUE TITLE
treboR's GTA V/RageMP Bugs

### ISSUE THREAD

**Issue by treboR-0** [ragemultiplayer/issues-archived#89]
GTA V/RageMP Fixes... "A list requested by George." seen or not seen on (W10 1607 GTAV 1.36) 170123

1) BORED CAMERA: When the player Camera is idle it looks around after 1 minute. Turn Off.
2) SPECTATE PLAYER: In order to help users you need to be able to see what i going on, Attach Camera to player regardless of in car or out, with full mouse rotation and near infinite mouse wheel scroll zoom in and out. Follows player through death/respawn. And If this is your doing, and not ours to do, Defaults to next or newest player if former player quits. And please not just using built in player cameras.
3) SET OBJECT DISTANCE DRAW: mp.players.forEach(_player => { _player.invoke(hashes.SET_ENTITY_LOD_DIST, object, 200); }); has no function set to 20 or 2000 does not change the draw distance, believed to be due to 1.36 1.37 hash differences, an in client universal hash translator was mentioned.
4) ARMOR: player.armour = 100; has no function 1.36
5) HEALTH: player.health = 100; has no function 1.36
6) VEHICLE SYNC: When following a vehicle they sometimes appear to move at an angle that is not expected based on the direction of the vehicle and real world known ways a car moves. the cars scuttle sideways like a crab without rotating properly.
7) VEHICLE SYNC: "repairs" vehicle locations where a vehicle moves to a location then moves to a slightly different position based on more accurate data but not related to in game player based action. Probably just a side effect of movement prediction and smoothing? MTA has yet to solve this problem either.
8) EVENTS: https://wiki.multitheftauto.com/wiki/CancelEvent
9) HANDLING: https://wiki.multitheftauto.com/wiki/SetVehicleHandling there was talk of a clientside api or atleast a great wiki explaining these existant features?
10) CAMERA: https://wiki.multitheftauto.com/wiki/SetCameraTarget, position, smooth follow
11) BLIPS: Vehicle Blips, unique idea. I like the idea. But needs proximity settings or server on off switch. Or distinguishing color so players can tell if occupied or not.
12) COLLISION: https://wiki.multitheftauto.com/wiki/SetElementCollisionsEnabled
13) REMOVE OBJECT: https://wiki.multitheftauto.com/wiki/RemoveWorldModel - note the global remove on this one. one hash, one perimeter, one distance, removes all objects with one command.
14: ALPHA: https://wiki.multitheftauto.com/wiki/SetElementAlpha

Otherwise surprising mod. Went straight to the top of my list. Glad I found you. You weren't found easily either. You need a much great world presence. client is solid, server is capable. You're largely ready for such.

**Comment by treboR-0** [comment]
and.... player.putIntoVehicle(player.veh, 0); this has no function.

**Comment by treboR-0** [comment]
and a file log...

console.log("OK");
consoleFile.log("BETTER");

**Comment by treboR-0** [comment]
"model": (player, args) =>
	{
		player.model = mp.joaat(args[1]);
		player.giveWeapon(game.weapons, 1000);
	},
No function.

**Comment by treboR-0** [comment]
W10 1.36 Hotfix 3

**Comment by treboR-0** [comment]
When the Master Server drops or whatever happened a few hours earlier. Servers running previously do not re announce when the Master comes back online.

**Comment by treboR-0** [comment]
I commonly rename old files as I update, like:  cp.nut to cp_o.nut for example. This allows me to downgrade in an instant if there is an unforseen problem. Doing so in RageMP loaded both named files automatically. loading some features, not others. very confusing.

**Comment by treboR-0** [comment]
Player blips appear to be IP related rather than player id. and a second player on the same ip appears to cause sync troubles to the first. namely a player setting player position like a teleport on the same ip as another can pull the other out of their vehicle, or cause it to desync as coop use to do all the time leaving two instances of the vehicle... maybe. only seen once.

**Comment by treboR-0** [comment]
player.position = new Vector3(x, y, z); may only function if you first teleport somewhere near. then far off positions which previously didn't function become accessible otherwise it teleports to same position.

And second machine mentioned above is: Win 7 Home Pro, GTA V 1.36 hot fix 3 on a LAN but connected through my remote VPS server

**Comment by treboR-0** [comment]
4 door cars are only 2 passenger. and people keep calling me TREBORO because the name tag is in all capitals and 0 adds to treboR to prevent nick collision? but should also add a - if you cant just say nick name in use, like treboR-0, for none ragemptester nicks.

**Comment by treboR-0** [comment]
on passenger exit monster, client freeze crash.

nick name is not stored must be reset each time

custom direct connect ip is not stored must be entered each time. favorites list?

**Comment by treboR-0** [comment]
"playername joined the server" message is located where?

George - it's hardcoded into client

add a switch to mute this message on join part quit

**Comment by treboR-0** [comment]
Distance between lines in chat box is very large if the chat gets busy not many lines can be there for long. Make distance between lines changeable, preferably by line-height:10px;

Chat box Input area is very large. and opacity/alpha could be more see through.

**Comment by treboR-0** [comment]
Passenger vehicle entry, push g. has no abort action from contrary motion like push f. if you push g to enter a vehicle that native movement cannot reach, like up a step higher than walking can climb, it will try to enter endlessly with no possible abort except to spawn a vehicle of your own, after which g does not function.

**Comment by treboR-0** [comment]
"RAGE MP: Connection lost. Reconnecting..." would be nice if this could be themed by the last server connected too.

**Comment by treboR-0** [comment]
Esc should close the chat box

**Comment by treboR-0** [comment]
once with me amd twice with a guy from italy clicking [p] pause has caused client to crash

**Comment by treboR-0** [comment]
most hours of the day master server is offline or unavailable. briefly when it is on players come and play.

**Comment by treboR-0** [comment]
MTA creates a backup server list each time you access the master list. Then consults the backup anytime the master isn't available.

**Comment by treboR-0** [comment]
player.seat reports incorrectly. often mistaking player for still being in the passenger seat. and always displaying driver or not driver as seat 255

**Comment by treboR-0** [comment]
turn off player bored animations. when the player stands it has "real life" animations like michael blowing snot out his nose or crotch grabbing, as casual real life... which it does every minute if you move then stand move then stand

**Comment by treboR-0** [comment]
With IV:MP in mind, toward the longevity and simple success of RageMP please make it function in total without your existence. Changable Master IP, Direct connect without Login. Local host without login. And only in Normal operation login, a UI for a 5 year old. Open > Click > Play. Every needless wall cuts your user base in half.
Walls stop people from participating. I am on RageMP because nanos lost me to steamcmd. A Wall.

**Comment by treboR-0** [comment]
When first joining a server, if you lag, you can see the back of another NPC before your own server assigns a skin.

**Comment by treboR-0** [comment]
if possible reverse mouse wheel should also be reverse weapons while in car, not radio.

**Comment by treboR-0** [comment]
Crashes 1.36 0.1-hf3 (W10P) Idle Over Night Would not wake up. (7HP) Idle Over Night Would not wake up. GTA V alone idles days on end no issues.

**Comment by treboR-0** [comment]
Adjustable sync rates would be great

Your startup UI is all screen dominating in 800x600 but still usable.

**Comment by treboR-0** [comment]
Switch to remove flag or use a RageMP stock flag from your own server on Launch client server list.

**Comment by treboR-0** [comment]
please be sure player name joined the server, left the server is mutable in 0.2 :)

**Comment by treboR-0** [comment]
Toggles for the Chat, HUD, and any related on screen visibility on and off.

**Comment by treboR-0** [comment]
1.36 0.1.0.3 170128 Client (W10P) Idle Over Night Would not wake up. (7HP) Idle Over Night Would not wake up.
1.36 0.1.0.3 170129 Client (W10P) Idle Over Night Would not wake up. (7HP) Idle Over Night Would not wake up.

**Comment by treboR-0** [comment]
seat belts save lives.

**Comment by treboR-0** [comment]
return server.Name

**Comment by treboR-0** [comment]
1.36 0.1.0.3 170129 (7HP) Idle 5 minutes Screen up. Exit passenger seat. Pushed F. Client Not responding.
1.36 0.1.0.3 170130 (7HP) Idle 35 minutes Screen up. Client disappeared. (7HP) post chatting Screen up. idle one minute disappeared (7HP) Idle Over Night. Screen disappeared. (10P) Idle over night woke up fine. (10P) Idle and active 24 hours straight 1st time.

**Comment by treboR-0** [comment]
saw another players life meter extend from the black box all the way off screen to the right

**Comment by treboR-0** [comment]
the long life meter isnt a bug. I took 30 seconds of minigun to the head to kill him, the life meter slowly reduced as i did. player name anxiousbullets2

**Comment by treboR-0** [comment]
he says its not a mod or cheat but that he has simple trainer in his original directory, he believes it is a glitch
, but f4 wont open his tainer

**Comment by treboR-0** [comment]
other players see his long life bar too.

**Comment by treboR-0** [comment]
list all players

**Comment by treboR-0** [comment]
1.36 0.1.0.3 170131 (10P) Crashed failing to enter vehicle as passenger due to pathway blockage, while I was trying to find the abort entry keystrokes for failed passenger entry.
1.36 0.1.0.3 170128 (10P) Idle minimized Over Night Client disappeared. (7HP) Idle maximized Over Night Client disappeared.

**Comment by ragemp** [comment]
@treboR-0 
>1.36 0.1.0.3 170128 (10P) Idle minimized Over Night Client disappeared. (7HP) Idle maximized Over Night Client disappeared.

#62 
>player.seat reports incorrectly. often mistaking player for still being in the passenger seat. and always displaying driver or not driver as seat 255

Yes, you have to check if player.vehicle is not null at first.

>Distance between lines in chat box is very large if the chat gets busy not many lines can be there for long. Make distance between lines changeable, preferably by line-height:10px;

It's already changeable, check /ui/style/main.css.

>I commonly rename old files as I update, like: cp.nut to cp_o.nut for example. This allows me to downgrade in an instant if there is an unforseen problem. Doing so in RageMP loaded both named files automatically. loading some features, not others. very confusing.

By default RAGE MP loads only index.js file per each package.

**Comment by treboR-0** [comment]
thanks for these fixes.

**Comment by treboR-0** [comment]
Windows 7 reaches a point where it crashes a lot. then wont open anymore. Processes indicate some unknown process is occupying 25% of System idle processes. Reboot fixes. w!0 has not displayed this problem.

**Comment by treboR-0** [comment]
//from the launcher following the ddos protection...

Account Suspended
This Account has been suspended.
Contact your hosting provider for more information.

One wall fails to let users past and the entire dream dies. Perhaps not this time but someday? No ability to change master server. no local host, or direct connect. You've got to keep the avenues unblocked.

**Comment by treboR-0** [comment]
In launch browser "Downloading another one DLC" no speakum good english. "Downloading yet another DLC" you perhaps meant if to be funny, or "Downloading another DLC" if to be plain and misinformative.

**Comment by treboR-0** [comment]
My smaller machine, now on an identical build of W10 1607 crashes the client (suddenly disappears 5+ times per day) as much as it did previously when running Windows 7HP. Disregard former reports of 7 crashing a lot.

https://wiki.multitheftauto.com/wiki/BindKey ...very nice to have.

When being shot relentlessly at the spawn eventually you cant get up, then later the screen turns black. at this point you can /spawn and walk around but the screen remains black, and finally the loading spinner lower right screen spins endessly.

A few Players in separate gun fights started dropping guns like bullets as they fire, piles of guns 50 to 100 guns fall to their feet, seemed to be a rifle ak47 likely.

**Comment by treboR-0** [comment]
Crashes 1-2 a day on my i7-4790 W10-1607 8GB GTX760-2GB, 5-10 per day on my i3-3225 W10-1607 8GB GTX560-1GB, both of which have run GTA V for one week without crashing.

My larger machine crashes the client every time I create a new video file using Vegas 14. It does not when running GTA V in sp.

Please be certain client stability of 0.2 approaches that of your server, before you release, bad client, great server, which has been faultless to date. Great Work.

**Comment by treboR-0** [comment]
Requested by George. My Crash reports. I know the importance of saying how each happened. But the vast majority of which occurred with the client sitting idle. I'm on my server 24/7 idle a lot, one minute the client is there, the next it is gone, the majority of which result in no windows error or "searching for the solution" box. The majority the client disappears, closes without report.

[i3-debug-0.1.0.3-Win10-1607-treboR-170209.log.txt](https://github.com/ragemultiplayer/issues/files/766506/i3-debug-0.1.0.3-Win10-1607-treboR-170209.log.txt)
[i7-debug-0.1.0.3-Win10-1607-treboR-170209.log.txt](https://github.com/ragemultiplayer/issues/files/766507/i7-debug-0.1.0.3-Win10-1607-treboR-170209.log.txt)

**Comment by treboR-0** [comment]
On a busy server [loading game models...] in the lower right hand corner of the screen, flashes continually every 2-10 seconds, it is visually fatiguing, it should disappear.
